File: backend/MasApi/Models/Customer.cs
using System.ComponentModel.DataAnnotations;
using System.Runtime.InteropServices;
using Microsoft.EntityFrameworkCore;

namespace MasApi.Models
{
    [Index(nameof(Email), IsUnique = true)]
    public class Customer: BaseModel
    {
        [Key]
        public required Guid Id { get; set; }
        public required string FirstName { get; set; }
        public required string LastName { get; set; }

        [EmailAddress]
        [MaxLength(450)]
        public required string Email { get; set; }
        public string? PhoneNumber { get; set; }
        public required Address Address { get; set; }

        public ICollection<Order>? Orders { get; set; }
    }
}

File: frontend/src/types/domain/customer.ts
import { Address } from './address';
import { Order2 } from '.';

export interface Customer {
    id: string; // UUID
    firstName: string;
    lastName: string;
    email?: string;
    phoneNumber?: string;
    address: Address;
    orders: Order2[] | null;
};

export interface CustomerTableData {
    id: string;
    lp: string;
    firstName: string;
    lastName: string;
    email: string;
    phone: string;
    address: string;
};
